src/moderation/moderation.service.ts

```diff
@@ -64,11 +64,29 @@ export class ModerationService {
         interactionAccountId: number,
         targetAccountId: number,
     ): Promise<boolean> {
-        const block = await this.db('blocks')
+        const accountBlock = await this.db('blocks')
             .where('blocker_id', targetAccountId)
             .andWhere('blocked_id', interactionAccountId)
             .first();
 
-        return block === undefined;
+        if (accountBlock !== undefined) {
+            return false;
+        }
+
+        const accountRow = await this.db('accounts')
+            .select('domain')
+            .where('id', interactionAccountId)
+            .first();
+
+        if (!accountRow) {
+            throw new Error(`Account missing ${interactionAccountId}`);
+        }
+
+        const domainBlock = await this.db('domain_blocks')
+            .where('blocker_id', targetAccountId)
+            .andWhere('domain', accountRow.domain)
```

Comment:
Is it better to do a join here, on the `accounts.domain`?

Reply:
I guess it would save a query right? something like:

```
select * from domain_blocks db
join accounts a on a.domain = db.domain
where db.domain = <domain>
and db.account_id = <blocked_account_id>
limit 1
```

but its probably not as explicit